python/xorbits/web/ui/src/Dashboard.js

```diff
@@ -86,73 +86,189 @@ class NodeInfo extends React.Component {
         )
       )
 
+    let target = Object.values(roleData)
+    let res = target.at('resource')
+    console.log(target)
+    console.log(res)
+
     const resourceStats = {
       cpu_total: gatherResourceStats('cpu_total'),
       cpu_avail: gatherResourceStats('cpu_avail'),
       memory_total: gatherResourceStats('memory_total'),
       memory_avail: gatherResourceStats('memory_avail'),
+      gpu_total: gatherResourceStats('gpu_total'),
+      gpu_avail: gatherResourceStats('gpu_avail'),
+      gpu_memory_total: gatherResourceStats('gpu_memory_total'),
+      gpu_memory_avail: gatherResourceStats('gpu_memory_avail'),
     }
+
+    //for all cases, we will at least have cpu information available.
     resourceStats.cpu_used = resourceStats.cpu_total - resourceStats.cpu_avail
     resourceStats.memory_used =
       resourceStats.memory_total - resourceStats.memory_avail
 
-    return (
-      <Table size="small">
-        <TableHead>
-          <TableRow>
-            <StyledTableCell style={{ fontWeight: 'bolder' }}>
-              Item
-            </StyledTableCell>
-            <StyledTableCell style={{ fontWeight: 'bolder' }}>
-              <Grid container>
-                <Grid>Value</Grid>
-              </Grid>
-            </StyledTableCell>
-          </TableRow>
-        </TableHead>
-        <TableBody>
-          <StyledTableRow>
-            <StyledTableCell>Count</StyledTableCell>
-            <StyledTableCell>
-              <Grid container>
-                <Grid>{Object.keys(this.state[this.nodeRole]).length}</Grid>
-              </Grid>
-            </StyledTableCell>
-          </StyledTableRow>
-          <StyledTableRow>
-            <StyledTableCell>CPU Info</StyledTableCell>
-            <StyledTableCell>
-              <Grid container>
-                <Grid xs={4}>Usage: {resourceStats.cpu_used.toFixed(2)}</Grid>
-                <Grid xs={8}>Total: {resourceStats.cpu_total.toFixed(2)}</Grid>
-              </Grid>
-            </StyledTableCell>
-          </StyledTableRow>
-          <StyledTableRow>
-            <StyledTableCell>Memory Info</StyledTableCell>
-            <StyledTableCell>
-              <Grid container>
-                <Grid xs={4}>
-                  Usage: {toReadableSize(resourceStats.memory_used)}
+    //in case that we have no gpu in use.
+    if (resourceStats.gpu_memory_total === undefined) {
```

Comment:
Just ``if (resourceStats.gpu_memory_total)`` is fine for javascript.

Reply:
Why this has to be ``=== undefined`` ?

Reply:
resolved in latest version